### PR TITLE
Story/csi 750 snap deployment 1 17

### PR DESCRIPTION
**Pull request by igorgonibm**
Snapshots support

### CODE REVIEW

File: deploy/olm-catalog/ibm-block-csi-operator/2.0.0/ibm-block-csi-operator.v2.0.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+              },
+              {
+                "name": "csi-attacher",
+                "repository": "registry.redhat.io/openshift4/ose-csi-external-attacher",
+                "tag": "latest",
+                "imagePullPolicy": "IfNotPresent"
+              },
+              {
+                "name": "csi-snapshotter",
+                "repository": "quay.io/k8scsi/csi-snapshotter",
```
[the remark this answers is not in the document]
Comment (igorgonibm) [reply deploy/olm-catalog/ibm-block-csi-operator/2.0.0/ibm-block-csi-operator.v2.0.0.clusterserviceversion.yaml:95]: Only here or in other places too?

File: pkg/config/settings.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ const (
 	CSIProvisionerImage         = "quay.io/k8scsi/csi-provisioner:v1.3.0"
 	CSIAttacherImage            = "quay.io/k8scsi/csi-attacher:v1.2.1"
 	CSILivenessProbeImage       = "quay.io/k8scsi/livenessprobe:v1.1.0"
+	CSISnapshotterImage      	= "quay.io/k8scsi/csi-snapshotter:v2.1.0"
```
[the remark this answers is not in the document]
Comment (igorgonibm) [reply pkg/config/settings.go:24]: Yes. Used Ctrl+Alt+L ad code scanning passed

File: deploy/olm-catalog/ibm-block-csi-operator/2.0.0/ibm-block-csi-operator.v2.0.0.clusterserviceversion.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (igorgonibm) [reply deploy/olm-catalog/ibm-block-csi-operator/2.0.0/ibm-block-csi-operator.v2.0.0.clusterserviceversion.yaml]: done

File: deploy/olm-catalog/ibm-block-csi-operator-community/1.2.0/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+              },
+              {
+                "name": "csi-attacher",
+                "repository": "quay.io/k8scsi/csi-attacher",
+                "tag": "v1.2.1",
+                "imagePullPolicy": "IfNotPresent"
+              },
+              {
+                "name": "livenessprobe",
+                "repository": "registry.redhat.io/openshift4/ose-csi-livenessprobe",
```
Comment (oriyarde) [inline deploy/olm-catalog/ibm-block-csi-operator-community/1.2.0/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml:95]: for community, use the 2.1.0 `quay.io/k8scsi/csi-snapshotter` (which you previously put in the Red Hat CSV)

File: deploy/olm-catalog/ibm-block-csi-operator/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,679 @@
...
+              },
+              {
+                "name": "csi-attacher",
+                "repository": "registry.redhat.io/openshift4/ose-csi-external-attacher",
+                "tag": "latest",
+                "imagePullPolicy": "IfNotPresent"
+              },
+              {
+                "name": "csi-snapshotter",
+                "repository": "registry.access.redhat.com/openshift4/ose-csi-external-snapshotter-rhel7",
```
Comment (oriyarde) [inline deploy/olm-catalog/ibm-block-csi-operator/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml:95]: please use the [correct registry hostname](https://github.com/IBM/ibm-block-csi-operator/pull/69#discussion_r415096646)

File: deploy/olm-catalog/ibm-block-csi-operator-community/1.2.0/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml (hunk 3 above)
Comment (oriyarde) [inline deploy/olm-catalog/ibm-block-csi-operator-community/1.2.0/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml:95]: for community, use the 2.1.0 `quay.io/k8scsi/csi-snapshotter` (which you previously put in the Red Hat CSV)

**Comment by oriyarde** [comment]
@igorgonibm do we have a ticket to align the rest of the sidecars for 1.17 (similar to [CSI-757](https://github.com/IBM/ibm-block-csi-operator/pull/57) for 1.16)?
in addition, could we delete this branch?

File: deploy/olm-catalog/ibm-block-csi-operator-community/1.2.0/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,679 @@
...
+    - IBM
+    - BlockStorage
+    - CSI
+  version: 1.2.0
+  replaces: ibm-block-csi-operator.v1.0.0
+  maturity: stable
+  maintainers:
+    - name: Guang Jiong Lou
+      email: luogj@cn.ibm.com
+  minKubeVersion: 1.14.1
```
Comment (oriyarde) [inline deploy/olm-catalog/ibm-block-csi-operator-community/1.2.0/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml:327]: we should remember to revise this field each time we add a new csv

File: deploy/olm-catalog/ibm-block-csi-operator-community/1.2.0/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,679 @@
...
+              {
+                "name": "csi-attacher",
+                "repository": "quay.io/k8scsi/csi-attacher",
+                "tag": "v1.2.1",
+                "imagePullPolicy": "IfNotPresent"
+              },
+              {
+                "name": "csi-snapshotter",
+                "repository": "quay.io/k8scsi/csi-snapshotter",
+                "tag": "2.1.0",
```
Comment (oriyarde) [inline deploy/olm-catalog/ibm-block-csi-operator-community/1.2.0/ibm-block-csi-operator.v1.2.0.clusterserviceversion.yaml:96]: missing `v` before version number

File: deploy/role.yaml [hunk 7]
```diff
@@ -175,3 +175,43 @@ rules:
...
   verbs:
   - '*'
+- apiGroups:
+  - snapshot.storage.k8s.io
+  resources:
+  - volumesnapshotclasses
+  verbs:
+  - get
+  - watch
+ - list
```
Comment (oriyarde) [inline deploy/role.yaml:185]: I think there is under-indentation here.
how do we know this yaml is valid?

**Pull request merged**